src/MessagePack.UnityClient/Assets/Scripts/MessagePack/Internal/Sequence`1.cs

```diff
@@ -228,6 +230,7 @@ private SequenceSegment GetSegment(int sizeHint)
             else
             {
                 sizeHint = Math.Max(this.MinimumSpanLength, sizeHint);
+                sizeHint = Math.Min(MaxHintSize, sizeHint);
```

Comment:
We cannot simply reduce `sizeHint`. [The contract for `IBufferWriter<T>.GetSpan`](https://docs.microsoft.com/en-us/dotnet/api/system.buffers.ibufferwriter-1.getspan?view=netcore-3.1) is that it provide at _least_ the size that it is asked for. So throwing is preferable to returning a smaller buffer.

Reply:
Yeah, that's why I was not sure it's the right place. I had a thought that by "set a reasonable cap on how large the arg is to GetSpan" you meant that it should be limited on Sequence side.
I did the change. Looked at other usages and found out that Lz4Block compression is not able to work with such big files (it is not serialized) and Lz4BlockArray is working just fine.
I guess we cannot do much with Lz4Block because of binary compatibility.